libhwmon/wrapadl.cpp

```diff
@@ -188,6 +206,21 @@ int wrap_adl_get_fanpcnt(wrap_adl_handle *adlh, int gpuindex, unsigned int *fanp
 	return 0;
 }
 
+int wrap_adl_get_power_usage(wrap_adl_handle *adlh, int gpuindex, unsigned int* miliwatts)
+{
+	wrap_adlReturn_t rc;
+	if (gpuindex < 0 || gpuindex >= adlh->adl_gpucount){
+		return -1;
+	}
+	int* power = new int;
```

Comment:
Getting closer, but this leaks! Try this:

int wrap_adl_get_power_usage(wrap_adl_handle *adlh, int gpuindex, unsigned int* miliwatts)
{
        wrap_adlReturn_t rc;
        if (gpuindex < 0 || gpuindex >= adlh->adl_gpucount){
                return -1;
        }
        int power;
        rc = adlh->adl2Overdrive6CurrentPowerGet(adlh->context, adlh->phys_logi_device_id[gpuindex], 0, &power);
        *miliwatts = power * 3.90625;
        return rc;
}

Works!

Reply:
All right! I submitted your fix... I guess it's working now! It gives me very similar results though...

Reply:
Sure. Should be exactly the same result actually. Just more direct.